src/sdam/topology.ts

```diff
@@ -568,25 +568,36 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
     }
 
     options = { serverSelectionTimeoutMS: this.s.serverSelectionTimeoutMS, ...options };
-    this.client.mongoLogger.debug(
-      MongoLoggableComponent.SERVER_SELECTION,
-      new ServerSelectionStartedEvent(selector, this.description, options.operationName)
-    );
+    if (
+      this.client.mongoLogger.willLog(SeverityLevel.DEBUG, MongoLoggableComponent.SERVER_SELECTION)
+    ) {
```

Comment:
(nit): Would it be beneficial to create a function similar to `shouldEmitAndLogCommand` and move the `willLog` check from here? True that there is only one condition that we check, but maybe with the function name it will look shorter and easier to read code. Plus, if we need to change it later, we will need to modify only one function rather than multiple `if`'s in the code. Looks good as is, just caught my eye since we are doing it quite a number of times in this pr.

Reply:
I was thinking something like that, but there are two differences between server selection logging and command logging to consider here:

1. Unlike CLAM, Server selection logging has different severity levels for its logs (debug and info), so regardless we will have to pass something into the function
2. in `shouldEmitAndLogCommand`, we check multiple booleans, here we are just calling one function.